qunicorn_core/core/pilotmanager/ibm_pilot.py

```diff
@@ -301,6 +301,14 @@ def __map_runner_results_to_dataclass(
                     meta=metadata,
                 )
             )
+            result_dtos.append(
+                ResultDataclass(
+                    program=programs[i] if programs else None,
```

Comment:
Alternatively to `if programs` add an error (or assert) before this part to make sure that programs are indeed available.